server/routes/api.js

```diff
@@ -117,13 +117,13 @@ routes.get('/pairing/paired', (req, res) => {
 
 routes.get('/wishlist/:familyId/read', (req, res) => {
 
-  const familyId = req.params.familyId;
+  const FAMILY_ID = req.params.familyId;
   // This regex checks if the family ID matches the parameters for an object ID
   // 0-9 a-f and a length of 24
-  if ( !familyId.match(/[0-F]{24}/gi) ) {
+  if ( !FAMILY_ID.match(/[0-F]{24}/gi) ) {
     res.status(400).json({ message: 'Invalid family id'});
   } else {
-    Wishlist.find({family: familyId})
+    Wishlist.find({family: FAMILY_ID})
```

Comment:
Can we discuss the use of CONSTANTS and assigning variables with const? 

I'm not seeing the need to capitalize these.

Reply:
We decided to make constants caps a long *long* time ago... I remember having this discussion when we were talking code conventions and deciding what rules the linter would catch.

Reply:
Yeah I have an understanding that constants should be all caps. but I dont see this being a semantic CONSTANT. 

To my understanding I would use all caps constants for something like a reusable string that we pre-define going in to the app. like "ORGANIZATION_NAME" and we'd define it as "Adopt-A-Family" and then we can call the constant ORGANIZATION_NAME anywhere in our code and it would reference that.

But we are binding a variable with const so it cannot be re-bound again during this function.

Reply:
So are you saying that this shouldn't be a constant and you think it should be some other type of variable?
I have only used constants in my tests for defining end points so that would be an example of a 'non-global' reusable string

Reply:
I am saying the use of non-reassignable variables in javascript do not need to be all caps. The only time I've actually seen it used as ALL_CAPS is when the value assigned to it is literally constant. Like my example earlier, a website name, a string for an action type.

If the value assigned to that variable comes from a user, from the return of a function, or a relative value, and we just don't want that being reassigned, I have only seen camelCased. 

In your tests:

```
const PAIRING_ENDPOINT = '/api/pairing';
const WISHLIST_ENDPOINT = '/api/wishlist';
const WISHLIST_CREATE_ENDPOINT = '/api/wishlist/addItem';
```

I agree this works because you're saying that these endpoints are going to be these and will never change. They're pre-defined and not a result of the frontend passing information in; or the result of a function call.

Reply:
In this particular case, I think the line in question should be `Wishlist.find({family: familyId})` and all caps case is misleading as the value is not static.

Reply:
I hear you and I agree 50%. If we are only wanting this variable to be scoped here, why don't we use `let` since it's just a way to make `req.params.familyId` easier to read. In that case wouldn't a `let family = req.params.familyId` be the right way to go?
Would anyone else here care to chime in on this? Should we go with `const` or `let` and if so, should we use the `const` with camelCase since it's not a true constant as josh pointed out

Reply:
>is not static

Is probably a much more concise of saying what im saying.

Reply:
It's fine to be a const declaration with camel case.  You could even destructure like so: `const { familyId } = req.params;`

Reply:
destructuring is super cool

Reply:
^I like that a lot. Makes it very clear that we're pulling a "constant" from the request.